_studio/mfx_lib/decode/mjpeg/src/mfx_mjpeg_dec_decode.cpp

```diff
@@ -1534,6 +1534,40 @@ mfxStatus VideoDECODEMJPEGBase_HW::Init(mfxVideoParam *decPar, mfxFrameAllocRequ
     m_pMJPEGVideoDecoder->SetFourCC(request_internal->Info.FourCC);
     m_numPic = 0;
 
+    MFX_CHECK_WITH_ASSERT(!m_pCc, MFX_ERR_UNDEFINED_BEHAVIOR);
+    if (m_needVpp)
+    {
+        bool isD3DToSys = false;
+
+        if(m_vPar.IOPattern & MFX_IOPATTERN_OUT_SYSTEM_MEMORY)
+        {
+            isD3DToSys = true;
+        }
+        else if (m_vPar.IOPattern & MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
+        {
+            mfxExtOpaqueSurfaceAlloc *pOpaqAlloc = (mfxExtOpaqueSurfaceAlloc *)GetExtendedBuffer(m_vPar.ExtParam, m_vPar.NumExtParam, MFX_EXTBUFF_OPAQUE_SURFACE_ALLOCATION);
+            if (!pOpaqAlloc)
+                return MFX_ERR_INVALID_VIDEO_PARAM;
+
+            isD3DToSys = !((pOpaqAlloc->Out.Type & MFX_MEMTYPE_SYSTEM_MEMORY) == 0);
+        }
+
+        mfxVideoParam localParams = m_vPar;
```

Comment:
I thought that idea was to not create full copy of ```mfxVideoParam```

Reply:
I would like to clarify. Are you for 
```
-mfxStatus VideoVppJpegD3D9::Init(const mfxVideoParam *par)
+mfxStatus VideoVppJpegD3D9::Init(mfxU16 AsyncDepth, const mfxFrameInfo &FrameInfo, mfxU16 Rotation,
+                                 const mfxInfoVPP &vpp)
```
or more elegant solution possible?

Reply:
@Alexandr-Konovalov, @mgonchar 

Maybe it makes sense to have `VideoVppJpegD3D9` as part of `mfx_UMC_FrameAllocator_D3D_Converter`? Looks like this class do nothing w/ decoding and perform only conversion that is what `mfx_UMC_FrameAllocator_D3D_Converter` is devoted to. What do u think?